src/Utils/Quaternion.cpp

```diff
@@ -3,37 +3,41 @@
 
 #include <Eigen/Dense>
```

Comment:
Is that possible to add a test for the derivative function. I am concern by the changes in the order of the constructor where we swapped things around!
Thanks :D